packages/apollo-forest-run/src/forest/types.ts

```diff
@@ -57,6 +60,18 @@ export type IndexedForest = {
 export type Source = Readonly<SourceObject | SourceCompositeList>;
 export type Draft = SourceObject | SourceCompositeList;
 
+export type UpdateChunkMetadata = {
+  type: string;
+  depth: number;
+  siblings: number;
+  dirtyFields: Set<string>;
+};
+
+export type UpdateForestMetadata = {
+  totalUpdated: number;
+  operationCosts: Map<OperationDescriptor, UpdateChunkMetadata[]>;
```

Comment:
WDYT if we rename it to `updateStats`? Or if we return the map itself, we probably don't need this whole type.